lib/ide/parser/src/api.rs

```diff
@@ -3,9 +3,53 @@
 use crate::prelude::*;
 
 use ast::IdMap;
+use ast::HasRepr;
+use ast::HasIdMap;
 
 pub use ast::Ast;
 
+use serde::Deserialize;
+use serde::Serialize;
+
+
+// ============
+// == Module ==
+// ============
+
+/// Parsed file / module with metadata
+#[derive(Debug,Clone,Serialize,Deserialize)]
+pub struct ModuleWithMetadata {
```

Comment:
I dont like the name `ModuleWithMetadata`. This is like naming `OperatorWithLeftHandSideAndRightHandSide`. Metadata is part of the module by design. This should be nam,ed just `Module`.